requires.py

```diff
@@ -12,76 +12,108 @@
+    related_units
+)
 
-class KubeControlRequireer(RelationBase):
-    """Implements the kubernetes-worker side of the kube-control interface.
 
-    """
-    scope = scopes.GLOBAL
 
-    @hook('{requires:kube-control}-relation-{joined,changed}')
+class KubeControlRequirer(Endpoint):
+    """
+    Implements the kubernetes-worker side of the kube-control interface.
+    """
+    @when_any('{endpoint_name}-relation-joined',
+              '{endpoint_name}-relation-changed')
     def joined_or_changed(self):
-        """Set states corresponding to the data we have.
-
         """
-        conv = self.conversation()
-        conv.set_state('{relation_name}.connected')
+        Set states corresponding to the data we have.
+        """
+        set_flag(self.expand_name('endpoint.{endpoint_name}.available'))
         self.check_states()
 
-    @hook('{requires:kube-control}-relation-departed')
+    @when('{endpoint_name}-relation-departed')
     def departed(self):
-        """Remove states corresponding to the data we have.
-
         """
-        conv = self.conversation()
+        Remove states corresponding to the data we have.
+        """
         # Make sure we have valid states as long as we still have related
         # units. Once all units are gone, clear all states.
-        if hookenv.related_units():
+        if related_units():
             self.check_states()
         else:
-            conv.remove_state('{relation_name}.connected')
-            conv.remove_state('{relation_name}.dns.available')
-            conv.remove_state('{relation_name}.auth.available')
-            conv.remove_state('{relation_name}.cluster_tag.available')
-            conv.remove_state('{relation_name}.registry_location.available')
+            clear_flag(
+                self.expand_name(
+                    'endpoint.{endpoint_name}.connected'))
+            clear_flag(
+                self.expand_name(
+                    'endpoint.{endpoint_name}.dns.available'))
+            clear_flag(
+                self.expand_name(
+                    'endpoint.{endpoint_name}.auth.available'))
+            clear_flag(
+                self.expand_name(
+                    'endpoint.{endpoint_name}.cluster_tag.available'))
+            clear_flag(
+                self.expand_name(
+                    'endpoint.{endpoint_name}.registry_location.available'))
 
     def check_states(self):
-        """Toggle states based on available data.
-
         """
-        conv = self.conversation()
+        Toggle states based on available data.
+        """
         if self.dns_ready():
-            conv.set_state('{relation_name}.dns.available')
+            set_flag(
+                self.expand_name(
+                    'endpoint.{endpoint_name}.dns.available'))
         else:
-            conv.remove_state('{relation_name}.dns.available')
+            clear_flag(
+                self.expand_name(
+                    'endpoint.{endpoint_name}.dns.available'))
```

Comment:
Not related to your change specifically, but these could be tidied up a bit using [`toggle_flag`](https://charmsreactive.readthedocs.io/en/latest/charms.reactive.flags.html#charms.reactive.flags.toggle_flag)